web/src/beta/features/PluginPlayground/Code/hook.ts

```diff
@@ -0,0 +1,125 @@
+import Visualizer from "@reearth/beta/features/Visualizer";
+import { useNotification } from "@reearth/services/state";
+import * as yaml from "js-yaml";
+import { ComponentProps, useCallback, useState } from "react";
+
+import { WidgetLocation } from "../../Visualizer/Crust/Widgets/types";
+import { FileType } from "../Plugins/constants";
+
+type Widgets = ComponentProps<typeof Visualizer>["widgets"];
+
+type ReearthYML = {
+  id: string;
+  name: string;
+  version: string;
+  extensions?: {
+    id: string;
+    type: string;
+    name: string;
+    description: string;
+    widgetLayout?: {
+      defaultLocation: {
+        zone: WidgetLocation["zone"];
+        section: WidgetLocation["section"];
+        area: WidgetLocation["area"];
+      };
+    };
+  }[];
+};
+
+type Props = {
+  files: FileType[];
+};
+
+const getYmlJson = (file: FileType) => {
+  if (file.sourceCode === "") {
+    return { success: false, message: "YAML file is empty" } as const;
+  }
```

Comment:
Passing an empty string as a parameter to yaml.load does not throw an error; instead, it results in undefined. Therefore, handling for the case of an empty string is added here.